patches/minecraft/net/minecraft/block/Block.java.patch

```diff
@@ -1404,14 +1404,40 @@
 +         return null;
 +     }
 +
-+     /**
++    /**
 +     * Queries if this block should render in a given layer.
-+     * ISmartBlockModel can use {@link net.minecraftforge.client.MinecraftForgeClient#getRenderLayer()} to alter their model based on layer.
++     * @deprecated use {@link #getRenderLayers(IBlockState)}
 +     */
++    @Deprecated // TODO: remove (1.13)
 +    public boolean canRenderInLayer(IBlockState state, BlockRenderLayer layer)
 +    {
 +        return func_180664_k() == layer;
 +    }
++
++    /**
++     * Returns the render layers this block uses.
++     * Prefer calling {@link #getRenderLayers(IBlockState, IBlockAccess, BlockPos)} where possible.
++     */
++    public Set<BlockRenderLayer> getRenderLayers(IBlockState state)
```

Comment:
Why is this one necessary? Can you just keep the one below?

Reply:
It's here because people seemed to want it, and I didn't have a particularly strong disagreement and didn't want to hold up the PR arguing about it.